client/src/search.tsx

```diff
@@ -175,76 +200,76 @@ export class SearchWidgetClass extends React.Component {
     this.fuzzySearcher = new FuzzySearch(urisSorted);
   };
 
-  searchHandler = (event) => {
-    this.setState({ q: event.target.value }, this.updateSearch);
-  };
-
   updateSearch = () => {
-    const q = this.state.q.trim();
-    if (!q) {
+    const { value, onValueExistsChange } = this.props;
+    if (!value) {
       if (this.state.showSearchResults) {
         this.setState({ showSearchResults: false });
       }
+      return;
     } else if (!this.index) {
       // This can happen if the initialized hasn't completed yet or
       // completed un-successfully.
       return;
-    } else {
-      // The iPhone X series is 812px high.
-      // If the window isn't very high, show fewer matches so that the
-      // overlaying search results don't trigger a scroll.
-      const limit = window.innerHeight < 850 ? 5 : 10;
-
-      if (q.startsWith("/") && !/\s/.test(q)) {
-        // Fuzzy-String search on the URI
-
-        if (q === "/") {
-          this.setState({
-            highlitResult: null,
-            searchResults: [],
-            showSearchResults: true,
-          });
-        } else {
-          const fuzzyResults = this.fuzzySearcher.search(q, { limit });
-          const results = fuzzyResults.map((fuzzyResult) => {
-            return {
-              title: this._map[fuzzyResult.needle].title,
-              uri: fuzzyResult.needle,
-              substrings: fuzzyResult.substrings,
-            };
-          });
-          this.setState({
-            highlitResult: results.length ? 0 : null,
-            searchResults: results,
-            showSearchResults: true,
-          });
-        }
-      } else {
-        // Full-Text search
-        const indexResults = this.index.search(q, {
-          limit,
-          // bool: "or",
-          suggest: true, // This can give terrible result suggestions
-        });
+    }
```

Comment:
looks like a giant change, it's mainly two things:

- early return in all the above cases, so that else doesn't need to be indented
- code reuse `if (results)`